### ISSUE TITLE
Interacting with the legends

### ISSUE THREAD

**Issue by PriPatel** [lightdash/lightdash#12867]
### Hover 
- [ ] If I hover over any label in the legend, it should isolate the series on the graph
- [ ] This should affect, compare to previous year, compare to another metric and any series when I have 'segmented by'

### Clicking
- [ ] If I click on any label in the legend it should hide the series on the chart, and the label should be greyed out in the legend. 
- [ ] If i click on a greyed out label in the legend, it should unhide the series from the chart

Basically same behaviour as normal saved charts in view mode
